exovetter/main.py

```diff
@@ -0,0 +1,61 @@
+
+#A sketch of an architecture. For discussion
+
+#Each vetting test is wrapped in a class that follows this
+#structure.
+class Vetter():
+    def __init__(self, **kwargs):
+        """kwargs stores the configuration parameters common to all TCEs
+        For example, for the Odd-even test, it might specify the signficance
+        of the depth difference that causes a TCE to fail
+        """
+        pass
+
+    def apply(self, tce, lightcurve):
```

Comment:
The constructor takes configuration arguments that are valid for every TCE examined (I'm thinking of things like thresholds to declare a TCE a false positive, which kind of lightcurve (sap or pdc) to examine, stuff like that). The apply() function is then called repeatedly for many TCEs. This approach makes sense because sometimes initialization of a vetter will be expensive. For example, the LPP vetter will have to load lots of data from disk, an operation we'd like to do only once. 

This approach scales down to single object science with only minimal overhead. Instead of calling `Lpp(tce, lightcurve, **config)`, you call `Lpp(**config).apply(tce, lightcuve)`. Not as elegant sure, but a small price to pay of amortizing the initialization cost. 

The tce and lightcurve are separate arguments because there isn't a one-to-one mapping between TCEs and lightcurves -- two TCEs can belong to the same star.

The name apply is inherited from a different codebase, and I agree it's not the best choice here. I'm tempted to make this a `__call__()` method, so that you could call `Lpp(**config)(tce, lightcurve)  but that's probably more confusing. I'm open to suggestions for a better name

Reply:
How about run() or calculate() instead of apply?

You are absolutely right that we don't want LPP to have to reload the large model each time it runs. There may be other metrics we develop that are like this as well.  This way of doing things naturally avoids that overhead.

Reply:
It almost sounds like we're training a model? In ML, there is the concept of `fit` and `predict`. Fitting data into a model so that it can be trained to predict an output. I'd be okay with changing the method name to `fit` if that is a good _fit_ for us in this package

Alternatively, expanding the name is considered pythonic. Rather than `apply`, it could be `add_tec_and_lightcurve_to_model`

Reply:
I am okay with anything more descriptive than `apply`. 😉  `astropy.modeling` and `synphot` both use `__call__`, so they won't be confusing for me, but extra documentation is definitely needed so users understand what they do if you want to go that route.

Reply:
LPP has fit and predict steps, but it's unusual in that sense. I think run() makes the most sense, and I'll have __call__() call run()